class.jetpack.php

```diff
@@ -486,6 +486,8 @@ private function __construct() {
 			Jetpack_Heartbeat::init();
 		}
 
+		add_action( 'rest_authentication_errors', array( $this, 'wp_json_authenticate' ) );
```

Comment:
Worth noting that even though `rest_authentication_errors` is the official way to do API authentication, it has a significant drawback.  If authentication fails, the request is never matched against an endpoint.

For WP.com this meant that we couldn't have per-endpoint monitoring for requests that failed auth.  To get around this, I decided to use a combination of `determine_current_user` and `rest_request_before_callbacks` for authentication.

I'm not sure if this is a concern for Jetpack or not, but wanted to note my experiences with it.